pkg/lib/serverTime.js

```diff
@@ -489,13 +630,98 @@
                         <td>
                             <Button variant="secondary" onClick={event => remove_server(event, i)}
                                     icon={ <CloseIcon /> }
-                                    isDisabled={i == custom_ntp.servers.length - 1} />
+                                    isDisabled={servers.length == 1 && servers[0] == ""} />
                         </td>
                     </tr>))
                 }
             </tbody>
         </table>);
 
+    return (
+        <Validated errors={errors} error_key="ntp_servers">
+            <div id="systime-ntp-servers-row">
+                <div id="systime-ntp-servers">
+                    { server_rows }
+                </div>
+            </div>
+        </Validated>);
+};
+
+const StaticServerList = ({ servers, change, map }) => {
+    function label(name) {
+        if (map && map[name])
+            return map[name];
+        else
+            return name;
+    }
+    return <>
+        { servers.map(s => <Checkbox key={s.name}
+                                            id={"cb-" + s.name}
+                                            label={label(s.name)}
+                                            isChecked={s.enabled}
+                                            onChange={val => { s.enabled = val; change() }} />)
+        }
+    </>;
+};
+
+const TimesyncdConfig = ({ state, errors, change }) => {
+    return <>
+        <Checkbox id="cb-specific"
+                         label={_("Use specific time servers instead of the standard ones")}
+                         isChecked={state.enabled}
+                         onChange={val => { state.enabled = val; change() }} />
+        { state.enabled &&
+        <DynamicServerList servers={state.servers} errors={errors} change={change} />
+        }
+    </>;
+};
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17992-20230113-125341-c8c3ecb6-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)